File: solutions/go/rotational-cipher/2/rotational_cipher.go
package rotationalcipher

import "unicode"

func RotationalCipher(plain string, shiftKey int) string {
    runes := make([]rune, len(plain))
    for i, r := range plain {
        rcode := int(r)
        if unicode.IsLetter(r) {
            base := 0
            switch {
            case 'a' <= r && r <= 'z':
                base = int('a')
			case 'A' <= r && r <= 'Z':
                base = int('A')
            }
            rcode = ((int(r) - base + shiftKey) % 26) + base
        } 
        runes[i] = rune(rcode)
    }
    return string(runes)
}

File: solutions/csharp/rotational-cipher/2/RotationalCipher.cs
public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        char Rotate(char c)
        {
            if (!char.IsLetter(c)) return c;
            int offset = char.IsLower(c) ? 'a' : 'A';
            return (char)(offset + ((c - offset + shiftKey) % 26));
        }

        return new([.. text.Select(Rotate)]);
    }
}